File: day2/day2.cpp
#include <iostream>
#include <fstream>
#include <algorithm>
#include <string>
using namespace std;

int main(int argc, char* argv[]) {
    // Create keypad and coordinates
    int x = 1, y = 1;
    int keypad[3][3];
    for (int r = 0; r < 3; r++) {
        for (int c = 0; c < 3; c++) {
            keypad[c][r] = 1 + c + (r * 3);
        }
    }

    string line;
    ifstream file("C:\\Users\\Phuc Tran\\Desktop\\Advent-of-Code\\day2\\input.txt");
    while (getline(file, line)) {
        for (int i = 0; i < line.length(); i++) {
            switch (line[i]) {
            case 'R': x = min(2, x + 1); break;
            case 'L': x = max(0, x - 1); break;
            case 'D': y = min(2, y + 1); break;
            case 'U': y = max(0, y - 1); break;
            }
        }
        cout << keypad[x][y];
    }
    cout << endl;
}

File: day2/day2.py
# Initialize the keypad and starting position
keypad = [[1, 2, 3],
          [4, 5, 6],
          [7, 8, 9]]
r, c = 1, 1

file = open("input.txt")
for line in file:
	# Update current position based on character read
	for char in line:
		if (char == 'U'):
			r = max(0, r - 1)
		elif (char == 'D'):
			r = min(2, r + 1)
		elif (char == 'L'):
			c = max(0, c - 1)
		elif (char == 'R'):
			c = min(2, c + 1)

	# Print the new position after reading the current line
	print(keypad[r][c])
file.close()